test/e2eexternal/setup.sh

```diff
@@ -0,0 +1,165 @@
+
+echo "Running with: globalnet=${globalnet@Q}, deploytool=${deploytool@Q}, deploytool_broker_args=${deploytool_broker_args@Q}, deploytool_submariner_args=${deploytool_submariner_args@Q}, cluster_settings=${cluster_settings@Q}, timeout=${timeout}, image_tag=${image_tag}, cable_driver=${cable_driver}, service_discovery=${service_discovery}"
+
+set -em -o pipefail
+
+source ${SCRIPTS_DIR}/lib/debug_functions
+source ${SCRIPTS_DIR}/lib/utils
+source ${SCRIPTS_DIR}/lib/deploy_funcs
+source test/e2eexternal/utils
+
+# Always source the shared cluster settings, to set defaults in case something wasn't set in the provided settings
+source "${SCRIPTS_DIR}/lib/cluster_settings"
+[[ -z "${cluster_settings}" ]] || source ${cluster_settings}
+
+### Functions ###
+function setup_external() {
+	# Create pseudo-ext docker network
+	docker network create --driver=bridge --subnet=${EXTERNAL_CIDRS} ${EXTERNAL_NET}
+
+	# Chose the first cluster as the one connected to external application
+	extcluster=$(get_external_cluster)
+
+	# Connect all nodes in the chosen cluster to pseudo-ext network
+	# Naming conventions of the nodes in kind are as described in
+	# https://github.com/kubernetes-sigs/kind/issues/1401#issuecomment-626108675
+	# (index starts from 1 and 1 is omitted).
+	num_control=1
+	num_worker=1
+	for node in ${cluster_nodes[${extcluster}]};do
+		if [ "${node}" = "control-plane" ];then
+			if [ ${num_control} -eq 1 ];then
+				docker network connect ${EXTERNAL_NET} ${extcluster}-${node}
+			else
+				docker network connect ${EXTERNAL_NET} ${extcluster}-${node}${num_control}
+			fi
+			num_control=$(( ${num_control} + 1))
+		elif [ "${node}" = "worker" ];then
+			if [ ${num_worker} -eq 1 ];then
+				docker network connect ${EXTERNAL_NET} ${extcluster}-${node}
+			else
+				docker network connect ${EXTERNAL_NET} ${extcluster}-${node}${num_worker}
+			fi
+			num_worker=$(( ${num_worker} + 1))
+		fi
+	done
+
+	# Create external test container on pseudo-ext network
+	docker run --cap-add=NET_ADMIN --name ${EXTERNAL_APP} --network ${EXTERNAL_NET} \
+		-d registry.access.redhat.com/ubi7/ubi python -m SimpleHTTPServer 80
+
+	# Add extra routing in external app
+	docker exec -i ${EXTERNAL_APP} yum install -y iproute
+
+	# Get IP address for submariner gateway IP on external network
+	subm_gw_node_name="${extcluster}-worker"
+	subm_gw_ip=$(get_ip_on_container_network ${subm_gw_node_name} ${EXTERNAL_NET})
+
+	# Add route for cluster CIDRs to go via subm_gw_ip
+	for subnet in ${cluster_CIDRs[@]};do
+		docker exec -i ${EXTERNAL_APP} ip r add ${subnet} via ${subm_gw_ip}
+	done
+
+	# Add route for service CIDR to go via subm_gw_ip
+	for subnet in ${service_CIDRs[@]};do
+		docker exec -i ${EXTERNAL_APP} ip r add ${subnet} via ${subm_gw_ip}
+	done
+}
+
+function add_external_cidrs() {
+	extcluster=$(get_external_cluster)
+	cluster_CIDRs[${extcluster}]="${cluster_CIDRs[${extcluster}]},${EXTERNAL_CIDRS}"
+}
+
+function subctl_install_subm_with_subnet() {
```

Comment:
Is it possible to just import and use https://github.com/submariner-io/shipyard/blob/devel/scripts/shared/lib/deploy_operator#L49

?